doc/Type/independent-routines.pod6

```diff
@@ -1164,6 +1164,65 @@ assigning L<Empty|/type/Slip#index-entry-Empty-Empty>; for everything else,
 equivalent to assigning L<Nil|/type/Nil> or C<Empty> in the case of arrays or
 hashes, whose use is advised.
 
+=head1 Array manipulation
+
+Routines that manipulate arrays and other containers.
+
+=head2 sub pop
+
+Defined as:
+
+    proto sub pop($, *%)
```

Comment:
I see and appreciate what you are doing here. It's a good idea to move independent subs to where they belong. However, I see some problems, which are not so much in the content itself but in the conventions. First one is here, in the use of `proto`. In general, it's not something we do. There are just 4 proto declarations in the reference section:
```Raku
doc/Type/Range.pod6:    proto method int-bounds(|)
doc/Type/Signature.pod6:proto sub stuff(|) {*}
doc/Type/Signature.pod6:This is often used in C<proto> definitions (like C<proto foo (|) {*}>) to
doc/Type/independent-routines.pod6:proto sub EVAL($code where Blob|Cool|Callable, Str() :$lang = 'Raku',
```
you're adding a bunch (here and in the other files), which would amount as many as there are now. I'm not totally sure about the reason why, but I guess it's simply an implementation detail you don't really need to know to use the feature. So you might consider taking them out.

Reply:
I noticed that it is not a convention to document prototypes. My reasons for breaking this convention are as follows.

In rakudo/rakudo/issues/1739 it is stated that many core subs have been converted to multis to allow users to add their own candidates. In my opinion this means both the prototype and the core candidates should be documented so users can find out which signatures they can use for their own candidates. The `pop` prototype won't dispatch to a candidate with two positional parameters, for example.

The same reasoning applies to multi methods. Even if the multi implementation was meant as an optimization only (as appears to be the case with `Array`'s `push` and `append`), the user should know about the prototype and the existing candidates so they can make an informed decision about whether to add a candidate or override the method altogether.

Of course a prototype like `(|)` does not need to be documented. I documented the protos of methods `push` and `append` because of the [`is nodal`](https://docs.raku.org/routine/is%20nodal) trait that it applies to all candidates.

Raku is very extensible, and I think it's good if readers of the documentation become aware of it. Of course there's a trade-off. Beginners may just want to get the gist of a sub or method and may be scared of by too many technical details. That's the reason why I added the "Probably best thought of as" sections, but I'm not sure if they are a good idea.

Reply:
It's also a convention that conventions are not changed in a PR :-); they should be discussed first in an issue, and find the best solution for that. I would argue, for instance, that protos should be documented separately from specific instances, since what you want to document for it is how to create a new multi. Totally different use case to calling the routine, which is what's being documented here.

Reply:
Ah, I was not aware  of *that* convention and did not intend to violate it :-)

I will remove the protos from this PR and in due time create an issue if no one else does or did.

Reply:
We're surrounded by conventions :-) Thanks a lot.